### PR TITLE
TruLens 1.2.9

### PR DESCRIPTION
**Pull request by sfc-gh-chu**
# Description

[Memoize base endpoint creation for cost tracking](https://github.com/truera/trulens/pull/1659) @sfc-gh-chu

[add importlib resources to conda build](https://github.com/truera/trulens/pull/1662) @sfc-gh-chu

[Switch from SQL function to REST API backend for Cortex Complete - co…](https://github.com/truera/trulens/pull/1671/commits/a7eee56927e59cf08330fbd03a00dbc852600409) @sfc-gh-dhuang

[fix meta.yaml spacing](https://github.com/truera/trulens/pull/1663) @sfc-gh-chu

[Fix Cortex provider for tests.](https://github.com/truera/trulens/pull/1666) @sfc-gh-dkurokawa

[relax snowflake-ml-python version](https://github.com/truera/trulens/pull/1664) @sfc-gh-chu

[Fix encoding issues in dashboard](https://github.com/truera/trulens/pull/1668) @sfc-gh-chu

## Other details good to know for developers

Please include any other details of this change useful for _TruLens_ developers.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to
  not work as expected)
- [ ] New Tests
- [ ] This change includes re-generated golden test results
- [ ] This change requires a documentation update

----

> [!IMPORTANT]
> Update to version 1.2.9 with Snowflake and Cortex integration improvements, new Jenkinsfile for CI/CD, and test setup enhancements.
> 
>   - **Version Update**:
>     - Update version to `1.2.9` in `pyproject.toml` and `meta.yaml` across multiple modules.
>   - **Snowflake and Cortex Integration**:
>     - Replace Snowflake connection with `snowpark_session` in `Cortex` class in `provider.py`.
>     - Update `test_endpoints.py` and `test_snowflake_feedback_evaluation.py` to use `snowpark_session`.
>     - Set `Cortex.DEFAULT_SNOWPARK_SESSION` in `SnowflakeTestCase` in `snowflake_test_case.py`.
>   - **Testing and CI/CD**:
>     - Add `Jenkinsfile` for CI/CD pipeline with Docker setup and test execution.
>     - Add `test-snowflake` target in `Makefile` for Snowflake-specific tests.
>   - **Dependency Changes**:
>     - Add `snowflake-ml-python` to `requirements.optional.txt` and `server_side_evaluation_artifacts.py`.
>     - Comment out `dill` dependency in `src/core/pyproject.toml`.
>   - **Miscellaneous**:
>     - Update `meta.yaml` files for version consistency.
>     - Minor code refactoring and cleanup in `app.py` and `endpoint.py`.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=truera%2Ftrulens&utm_source=github&utm_medium=referral)<sup> for 06148714684601b580edbd28a40bb308a4ff80db. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

File: src/core/pyproject.toml [hunk 1]
```diff
@@ -33,7 +33,7 @@ numpy = [
   { version = ">=1.23.0", python = ">=3.9" },
 ]
 munch = "^2.5"
-dill = ">=0.3.8"
+# dill = ">=0.3.8"
```
Comment (sfc-gh-dhuang) [inline src/core/pyproject.toml:36]: why are we removing `dill`?

Comment (sfc-gh-chu) [reply src/core/pyproject.toml:36]: lol sorry bad commit

**Review by sfc-gh-dhuang — APPROVED** [review]
thanks for dilling with this

**Comment by sfc-gh-jreini** [comment]
@coreyhu can you merge and create a github release for this?

**Pull request merged**